dds/DCPS/XTypes/Utils.cpp

```diff
@@ -1381,10 +1381,17 @@ DDS::ReturnCode_t copy_member(
       DDS::DynamicData_var subsrc;
       get_rc = src->get_complex_value(subsrc, src_id);
       if (get_rc == DDS::RETCODE_OK) {
-        DDS::DynamicData_var subdest;
-        get_rc = dest->get_complex_value(subdest, dest_id);
-        if (get_rc == DDS::RETCODE_OK) {
-          set_rc = copy(subdest, subsrc);
+        DDS::DynamicType_var parent_dest_type = get_base_type(dest_type);
+        if (parent_dest_type->get_kind() == TK_UNION && dest_id != DISCRIMINATOR_ID) {
```

Comment:
I think it doesn't need to check for `DISCRIMINATOR_ID` because copying the discriminator won't go to this case.